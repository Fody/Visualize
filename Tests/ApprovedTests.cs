using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

[UseReporter(typeof(DiffReporter))]
#if(DEBUG)
[UseApprovalSubdirectory("results-debug")]
#else
[UseApprovalSubdirectory("results-release")]
#endif
[TestFixture]
public class ApprovedTests
{
    [Test]
    public void ClassWithProperties()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithProperties"));
    }

    [Test]
    public void ClassWithConst()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithConst"));
    }

    [Test]
    public void ClassWithIndexor()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIndexor"));
    }

    [Test]
    public void ClassWithDataAnnotations()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithDataAnnotations"));
    }

    [Test]
    public void ClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerable"));
    }

    [Test]
    public void ClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithICollection"));
    }

    [Test]
    public void GenericClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithIEnumerable`1"));
    }

    [Test]
    public void GenericClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithICollection`1"));
    }

    [Test]
    public void ClassWithIEnumerableNotAttributed()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerableNotAttributed"));
    }

    //[Test]
    //public void Sample()
    //{
    //    var sampleClassType = AssemblyWeaver.Assembly.GetType("AssemblyToProcess.ClassWithIEnumerable");
    //    var sample = Activator.CreateInstance(sampleClassType);
    //}
}