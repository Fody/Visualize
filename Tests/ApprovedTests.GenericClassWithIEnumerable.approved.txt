.assembly extern mscorlib
{
  .publickeytoken = (B7 7A 5C 56 19 34 E0 89 )                         // .z\V.4..
  .ver 4:0:0:0
}
.assembly extern System.Core
{
  .publickeytoken = (B7 7A 5C 56 19 34 E0 89 )                         // .z\V.4..
  .ver 4:0:0:0
}
.assembly AssemblyToProcess
{
  .custom instance void [mscorlib]System.Runtime.CompilerServices.RuntimeCompatibilityAttribute::.ctor() = ( 01 00 01 00 54 02 16 57 72 61 70 4E 6F 6E 45 78   // ....T..WrapNonEx
                                                                                                             63 65 70 74 69 6F 6E 54 68 72 6F 77 73 01 )       // ceptionThrows.
  // --- The following custom attribute is added automatically, do not uncomment -------
  //  .custom instance void [mscorlib]System.Diagnostics.DebuggableAttribute::.ctor(valuetype [mscorlib]System.Diagnostics.DebuggableAttribute/DebuggingModes) = ( 01 00 07 01 00 00 00 00 ) 
  .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilationRelaxationsAttribute::.ctor(int32) = ( 01 00 08 00 00 00 00 00 ) 
  .custom instance void [mscorlib]System.Runtime.Versioning.TargetFrameworkAttribute::.ctor(string) = ( 01 00 1A 2E 4E 45 54 46 72 61 6D 65 77 6F 72 6B   // ....NETFramework
                                                                                                        2C 56 65 72 73 69 6F 6E 3D 76 34 2E 30 01 00 54   // ,Version=v4.0..T
                                                                                                        0E 14 46 72 61 6D 65 77 6F 72 6B 44 69 73 70 6C   // ..FrameworkDispl
                                                                                                        61 79 4E 61 6D 65 10 2E 4E 45 54 20 46 72 61 6D   // ayName..NET Fram
                                                                                                        65 77 6F 72 6B 20 34 )                            // ework 4
  .hash algorithm 0x00008004
  .ver 0:0:0:0
}
.module AssemblyToProcess.dll
.imagebase 0x00400000
.file alignment 0x00000200
.stackreserve 0x00100000
.subsystem 0x0003       // WINDOWS_CUI
.corflags 0x00000001    //  ILONLY
.custom (AssemblyToProcess.GenericClassWithIEnumerable`1) instance void [mscorlib]System.Diagnostics.DebuggerTypeProxyAttribute::.ctor(class [mscorlib]System.Type) = ( 01 00 52 41 73 73 65 6D 62 6C 79 54 6F 50 72 6F   // ..RAssemblyToPro
                                                                                                                                                                        63 65 73 73 2E 47 65 6E 65 72 69 63 43 6C 61 73   // cess.GenericClas
                                                                                                                                                                        73 57 69 74 68 49 45 6E 75 6D 65 72 61 62 6C 65   // sWithIEnumerable
                                                                                                                                                                        60 31 2B 3C 47 65 6E 65 72 69 63 43 6C 61 73 73   // `1+<GenericClass
                                                                                                                                                                        57 69 74 68 49 45 6E 75 6D 65 72 61 62 6C 65 3E   // WithIEnumerable>
                                                                                                                                                                        50 72 6F 78 79 00 00 )                            // Proxy..
.custom (AssemblyToProcess.GenericClassWithIEnumerable`1/'<GetEnumerator>d__0') instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [mscorlib]System.Diagnostics.DebuggerBrowsableState) = ( 01 00 03 00 00 00 00 00 ) 
.custom (AssemblyToProcess.GenericClassWithIEnumerable`1/'<GenericClassWithIEnumerable>Proxy') instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance !0 AssemblyToProcess.GenericClassWithIEnumerable`1/'<GetEnumerator>d__0'::'System.Collections.Generic.IEnumerator<T>.get_Current'()) instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (AssemblyToProcess.GenericClassWithICollection`1) instance void [mscorlib]System.Diagnostics.DebuggerTypeProxyAttribute::.ctor(class [mscorlib]System.Type) = ( 01 00 52 41 73 73 65 6D 62 6C 79 54 6F 50 72 6F   // ..RAssemblyToPro
                                                                                                                                                                        63 65 73 73 2E 47 65 6E 65 72 69 63 43 6C 61 73   // cess.GenericClas
                                                                                                                                                                        73 57 69 74 68 49 43 6F 6C 6C 65 63 74 69 6F 6E   // sWithICollection
                                                                                                                                                                        60 31 2B 3C 47 65 6E 65 72 69 63 43 6C 61 73 73   // `1+<GenericClass
                                                                                                                                                                        57 69 74 68 49 43 6F 6C 6C 65 63 74 69 6F 6E 3E   // WithICollection>
                                                                                                                                                                        50 72 6F 78 79 00 00 )                            // Proxy..
.custom (AssemblyToProcess.GenericClassWithICollection`1) instance void [mscorlib]System.Diagnostics.DebuggerDisplayAttribute::.ctor(string) = ( 01 00 2B 43 6F 75 6E 74 20 3D 20 7B 43 6F 75 6E   // ..+Count = {Coun
                                                                                                                                                 74 7D 20 7C 20 49 73 52 65 61 64 4F 6E 6C 79 20   // t} | IsReadOnly 
                                                                                                                                                 3D 20 7B 49 73 52 65 61 64 4F 6E 6C 79 7D 00 00 ) // = {IsReadOnly}..
.custom (method instance void AssemblyToProcess.GenericClassWithIEnumerable`1/'<GetEnumerator>d__0'::System.Collections.IEnumerator.Reset()) instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0') instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (AssemblyToProcess.GenericClassWithICollection`1/'<GenericClassWithICollection>Proxy') instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance object AssemblyToProcess.GenericClassWithIEnumerable`1/'<GetEnumerator>d__0'::System.Collections.IEnumerator.get_Current()) instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (AssemblyToProcess.ClassWithExistingAttributes) instance void [mscorlib]System.Diagnostics.DebuggerDisplayAttribute::.ctor(string) = ( 01 00 07 4E 6F 74 68 69 6E 67 00 00 )             // ...Nothing..
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [mscorlib]System.Diagnostics.DebuggerBrowsableState) = ( 01 00 03 00 00 00 00 00 ) 
.custom (method instance void AssemblyToProcess.GenericClassWithIEnumerable`1/'<GetEnumerator>d__0'::.ctor(int32)) instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (AssemblyToProcess.ClassWithICollection) instance void [mscorlib]System.Diagnostics.DebuggerDisplayAttribute::.ctor(string) = ( 01 00 2B 43 6F 75 6E 74 20 3D 20 7B 43 6F 75 6E   // ..+Count = {Coun
                                                                                                                                        74 7D 20 7C 20 49 73 52 65 61 64 4F 6E 6C 79 20   // t} | IsReadOnly 
                                                                                                                                        3D 20 7B 49 73 52 65 61 64 4F 6E 6C 79 7D 00 00 ) // = {IsReadOnly}..
.custom (AssemblyToProcess.ClassWithICollection) instance void [mscorlib]System.Diagnostics.DebuggerTypeProxyAttribute::.ctor(class [mscorlib]System.Type) = ( 01 00 42 41 73 73 65 6D 62 6C 79 54 6F 50 72 6F   // ..BAssemblyToPro
                                                                                                                                                               63 65 73 73 2E 43 6C 61 73 73 57 69 74 68 49 43   // cess.ClassWithIC
                                                                                                                                                               6F 6C 6C 65 63 74 69 6F 6E 2B 3C 43 6C 61 73 73   // ollection+<Class
                                                                                                                                                               57 69 74 68 49 43 6F 6C 6C 65 63 74 69 6F 6E 3E   // WithICollection>
                                                                                                                                                               50 72 6F 78 79 00 00 )                            // Proxy..
.custom (AssemblyToProcess.ClassWithICollection/'<ClassWithICollection>Proxy') instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (AssemblyToProcess.ClassWithIEnumerable) instance void [mscorlib]System.Diagnostics.DebuggerTypeProxyAttribute::.ctor(class [mscorlib]System.Type) = ( 01 00 42 41 73 73 65 6D 62 6C 79 54 6F 50 72 6F   // ..BAssemblyToPro
                                                                                                                                                               63 65 73 73 2E 43 6C 61 73 73 57 69 74 68 49 45   // cess.ClassWithIE
                                                                                                                                                               6E 75 6D 65 72 61 62 6C 65 2B 3C 43 6C 61 73 73   // numerable+<Class
                                                                                                                                                               57 69 74 68 49 45 6E 75 6D 65 72 61 62 6C 65 3E   // WithIEnumerable>
                                                                                                                                                               50 72 6F 78 79 00 00 )                            // Proxy..
.custom (AssemblyToProcess.ClassWithIEnumerable/'<ClassWithIEnumerable>Proxy') instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [mscorlib]System.Diagnostics.DebuggerBrowsableState) = ( 01 00 03 00 00 00 00 00 ) 
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (AssemblyToProcess.ClassWithProperties) instance void [mscorlib]System.Diagnostics.DebuggerDisplayAttribute::.ctor(string) = ( 01 00 53 42 6F 6F 6C 65 61 6E 20 3D 20 7B 42 6F   // ..SBoolean = {Bo
                                                                                                                                       6F 6C 65 61 6E 7D 20 7C 20 4E 75 6C 6C 61 62 6C   // olean} | Nullabl
                                                                                                                                       65 20 3D 20 7B 4E 75 6C 6C 61 62 6C 65 7D 20 7C   // e = {Nullable} |
                                                                                                                                       20 4E 75 6D 62 65 72 20 3D 20 7B 4E 75 6D 62 65   //  Number = {Numbe
                                                                                                                                       72 7D 20 7C 20 53 74 72 69 6E 67 20 3D 20 7B 53   // r} | String = {S
                                                                                                                                       74 72 69 6E 67 7D 00 00 )                         // tring}..
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [mscorlib]System.Diagnostics.DebuggerBrowsableState) = ( 01 00 03 00 00 00 00 00 ) 
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (UNKNOWN_OWNER) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance !0 AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'::'System.Collections.Generic.IEnumerator<T>.get_Current'()) instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance void AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'::System.Collections.IEnumerator.Reset()) instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance object AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'::System.Collections.IEnumerator.get_Current()) instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance void AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'::.ctor(int32)) instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance string AssemblyToProcess.ClassWithExistingAttributes::get_String()) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance void AssemblyToProcess.ClassWithExistingAttributes::set_String(string)) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance int32 AssemblyToProcess.ClassWithProperties::get_Number()) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance void AssemblyToProcess.ClassWithProperties::set_Number(int32)) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance string AssemblyToProcess.ClassWithProperties::get_String()) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance void AssemblyToProcess.ClassWithProperties::set_String(string)) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance bool AssemblyToProcess.ClassWithProperties::get_Boolean()) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance void AssemblyToProcess.ClassWithProperties::set_Boolean(bool)) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance valuetype [mscorlib]System.Nullable`1<int32> AssemblyToProcess.ClassWithProperties::get_Nullable()) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance void AssemblyToProcess.ClassWithProperties::set_Nullable(valuetype [mscorlib]System.Nullable`1<int32>)) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance string AssemblyToProcess.ClassWithProperties::get_Internal()) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom (method instance void AssemblyToProcess.ClassWithProperties::set_Internal(string)) instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 